pymeasure/instruments/hp/hp437b.py

```diff
@@ -0,0 +1,949 @@
+        self.write("*CLS")
+
+    def preset(self):
+        """
+        Sets the power meter to a known state. Preset
+        conditions are shown in the following table.
+
+        .. list-table:: Preset values
+            :widths: 25 25
+            :header-rows: 1
+
+            * - Parameter
+              - Value/Condition
+            * - Frequency
+              - 50 MHz
+            * - Resolution
+              - 0.01 dB
+            * - Duty Cylce
+              - 1.000%, Off
+            * - Relative
+              - 0 dB, Off
+            * - Power Reference
+              - Off
+            * - Range
+              - Auto
+            * - Unit
+              - dBm
+            * - Low Limit
+              - -90.000 dBm
+            * - High Limit
+              - +90.000 dBm
+            * - Limit Checking
+              - Off
+            * - Trigger Mode
+              - Free Run
+            * - Group Trigger Mode
+              - Trigger with Delay
+            * - Display Function
+              - Display Enable
+        """
+        self.write("PR")
+
+    relative_mode_enabled = Instrument.control(
+        "SM", "RL%d",
+        """
+        Control the relative mode. In the relative mode the current measured power value will be
+        used as reference and any further reported value from :attr:`power` will refer to this.
+        """,
+        map_values=True,
+        values={True: 1, False: 0},
+        cast=int,
+        get_process=__getstatus(StatusMessage.RelativeModeStatus),
+        check_set_errors=True
+    )
+
+    measurement_unit = Instrument.measurement(
+        "SM",
+        """
+        Get the measurement unit the power meter is currently reporting the power values in.
+
+        Depends on: :attr:`relative_mode_enabled` and attr:`log_lin`
+
+        .. code-block:: python
+
+            instr.relative_mode_enabled = False
+            instr.log_lin = LogLin.Linear
+
+            print(instr.measurement_unit)
+            MeasurementUnit.Watts
+
+        """,
+        values={e: int(f.value) for (e, f) in zip(MeasurementUnit, MeasurementUnit)},
+        map_values=True,
+        get_process=__getstatus(StatusMessage.MeasurementUnits),
+    )
+
+    log_lin = Instrument.control(
+        "SM", "%s",
+        """
+        Control if the power meter displays or reports the power values in logarithmic or linear
+        units. See :attr:`measurement_unit` for further information.
+
+        .. code-block:: python
+
+            from pymeasure.instruments.hp.hp437b import LogLin
+
+            instr.relative_mode_enabled = False
+            instr.log_lin = LogLin.Linear
+        """,
+        validator=strict_discrete_set,
+        values=[e for e in LogLin],
+        set_process=lambda v: "LN" if int(v.value) == 0 else "LG",
+        get_process=__getstatus(StatusMessage.LinearLogStatus, lambda v: {int(f.value): e for (
+            e, f) in zip(LogLin, LogLin)}[v])
+    )
```

Comment:
what about using `log_enabled` instead?
That way you have linear as the default and you can enable/disable logarithmic options. That makes it more clearer for the code: `log_enabled = True`, for example.

You could name it also `logarithmic_display_enabled` or anything similar.

Reply:
I like the idea to adhere to the common code style in this repository. But I will do it the opposite and call if linear_display_enabled in the RF world logarithmic is way more common used.